### ISSUE TITLE
False positive unused_lifetimes warning with an async fn

### ISSUE THREAD

**Issue by Hawk777** [rust-lang/rust#86615]
Given [the following code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2fce0b89b3c5742798095a09811c56a7):

```rust
#![warn(unused_lifetimes)]

use std::borrow::Cow;

pub async fn do_thing(s: Cow<'_, str>) -> Cow<'_, str> {
	s
}
```

The current output is:

```
warning: lifetime parameter `'_` never used
 --> src/main.rs:5:47
  |
5 | pub async fn do_thing(s: Cow<'_, str>) -> Cow<'_, str> {
  |                              -----------------^^
  |                              |
  |                              help: elide the unused lifetime
  |
```

Removing the word `async` makes the warning disappear. I am currently running 1.52.1, and this also happens in 1.53.0 on the playground, as well as 1.55.0 nightly on the playground.

**Comment by tmandry** [comment]
Someone interested in fixing this _might_ find it helpful to look at #79689.

**Comment by Aaron1011** [comment]
This no longer warns as of the latest nightly.

**Comment by scottmcm** [comment]
Closing as the example in the OP no longer repros.

**Comment by Aaron1011** [comment]
Does this need a regresion test?
